src/main/java/net/minecraftforge/server/permission/nodes/PermissionNode.java

```diff
@@ -0,0 +1,121 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission.nodes;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+
+/**
+ * @see PermissionNode#PermissionNode(String, PermissionType, PermissionResolver, PermissionDynamic[])
+ * @param <T> PermissionType see {@link PermissionTypes}
+ */
+public class PermissionNode<T>
+{
+   private final String nodeName;
+   private final PermissionType<T> type;
+   private final PermissionResolver<T> resolver;
+   private final PermissionDynamic[] dynamics;
+
+   private String readableName;
+   private String description;
+
+   /**
+    * Calls {@link PermissionNode#PermissionNode(String, PermissionType, PermissionResolver, PermissionDynamic[])}
+    * with "namespace.path" as the first parameter
+    */
+   public PermissionNode(ResourceLocation nodeName, PermissionType<T> type, PermissionResolver<T> resolver, PermissionDynamic... dynamics)
+   {
+      this(nodeName.getNamespace() + "." + nodeName.getPath(), type, resolver, dynamics);
+   }
+
+   /**
+    * Calls {@link PermissionNode#PermissionNode(String, PermissionType, PermissionResolver, PermissionDynamic[])}
+    * with "modname.nodename" as the first parameter
+    */
+   public PermissionNode(String modName, String nodeName, PermissionType<T> type, PermissionResolver<T> resolver, PermissionDynamic... dynamics)
+   {
+      this(modName + "." + nodeName, type, resolver, dynamics);
+   }
+
+   /**
+    * @param nodeName The identifier of a node, recommended identifier structure: "modid.path.for.node"
+    * @param type type object for the PermissionNode, only use types in {@link PermissionTypes}, if you need more PR them
+    * @param resolver Default resolver for the permission, can but doesn't have to be used to by PermissionHandlers
+    * @param dynamics PermissionDynamic is a dynamic component for permission nodes, similar to BlockState Properties.
+    *                 They <strong>must</strong> be paassed into the constructor if you want to use them.
+    */
+   public PermissionNode(String nodeName, PermissionType<T> type, PermissionResolver<T> resolver, PermissionDynamic... dynamics)
+   {
+      this.nodeName = nodeName;
+      this.type = type;
+      this.dynamics = dynamics;
+      this.resolver = resolver;
+   }
+
+   /**
+    * <p>Allows you to set a human-readable name & description for your Permission.
+    * It is recommended to keep descriptions short, but there is no hard limit.</p>
+    *
+    * @param readableName a readable name for the PermissionNode
+    * @param description description for the PermissionNode
+    * @return itself with the new information set.
+    */
```

Comment:
I'd add that permission Handlers may use this information in game for something. And also a suggested max length, like at most 200 chars or something.

Reply:
How could we adjust this to allow translations?
Do we want to change to TextComponents, or just set translation keys here instead?